src/logics/stargate-v2/logic.swap-token.ts

```diff
@@ -0,0 +1,129 @@
+import { StargatePool__factory } from './contracts';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { getDestChainIds, getEndpointId, getMarkets, getPoolByTokenAddress, supportedChainIds } from './configs';
+import { getNativeToken } from '@protocolink/common';
+import { utils } from 'ethers';
+
+export type SwapTokenLogicTokenList = {
+  srcToken: common.Token;
+  destChainIds: number[];
+}[];
+
+export type SwapTokenLogicParams = core.TokenToTokenExactInParams<{
+  receiver: string;
+}>;
+
+export type SwapTokenLogicFields = core.TokenToTokenExactInFields<{
+  receiver: string;
+  fee: string;
+  lzTokenFee: string;
+}>;
+
+export type SwapTokenLogicOptions = Pick<core.GlobalOptions, 'account'>;
+
+export class SwapTokenLogic extends core.Logic implements core.LogicBuilderInterface {
+  static id = 'swap-token';
+  static protocolId = 'stargate-v2';
+  static readonly supportedChainIds = supportedChainIds;
+
+  async getTokenList() {
+    const tokenList: SwapTokenLogicTokenList = [];
+    const srcTokens = [];
+
+    const markets = getMarkets(this.chainId);
+    for (const market of markets) {
+      srcTokens.push(market.token);
+    }
+
+    // find destination chain ids
+    for (const srcToken of srcTokens) {
+      const destChainIds = getDestChainIds(this.chainId, srcToken);
+      tokenList.push({ srcToken, destChainIds });
+    }
+
+    return tokenList;
+  }
+
+  public async quote(params: SwapTokenLogicParams) {
+    const { input, tokenOut, receiver } = params;
+    const dstEid = getEndpointId(tokenOut.chainId);
```

Comment:
For the `tokenOut` prop in `quote`, only the `chainId` matters. Should we prevent users from passing incorrect tokens? and the same as `build` function

Reply:
I see. Let's decide if we need to provide destToken in token list first then we can move onto this one.